nana/nana.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"os"
+)
+
+func main() {
+
+	var fNames string
+	patcher := flag.NewFlagSet("sew", flag.ExitOnError)
+	saveAs := patcher.String("saveAs", "patched.go", "the file to write to disk")
+	patcher.StringVar(&fNames, "files", "", "the files to sew together")
+	remove := patcher.Bool("D", false, "delete files after patching")
+
+	deleter := flag.NewFlagSet("delete", flag.ExitOnError)
+	funcName := deleter.String("func", "", "function to patch -- example: -func nameOfFunc")
+	fileName := deleter.String("file", "", "the path to the file to delete -- example: -file path/to/file")
+
+	if len(os.Args) < 3 {
+		fmt.Println("not enough arguments passed")
+		os.Exit(51)
+	}
+
+	if os.Args[1] == "delete" {
+		deleter.Parse(os.Args[2:])
+		fmt.Println("deleting ", *fileName)
+		delete(*fileName, *funcName)
+	} else if os.Args[1] == "sew" {
+		patcher.Parse(os.Args[2:])
+		args := patcher.Args()
+		fmt.Println(*saveAs)
+		fmt.Println(args)
+		patch(*saveAs, *remove, args...)
+	}
+
+	fmt.Printf("command '%s' not recognized", os.Args[1])
+	os.Exit(50)
```

Comment:
this needs to always exit 50? there's no path out of this main that doesn't 

it also seems a bit potentially dodgy manually parse organize `os.Args` and also to be parsing arguments using flag.

Reply:
> this needs to always exit 50? there's no path out of this main that doesn't

right, ill switch this to a switch/case instead and have that be the default

> it also seems a bit potentially dodgy manually parse organize os.Args and also to be parsing arguments using flag.

hmm i wasn't sure if there's a better way to go about it - do you have any suggestion?

Reply:
I would use cobra or whatever your favorite CLI flag parsing library for this kind of thing.